### PR TITLE
Automated cherry pick of #121566: etcd: Update to version 3.5.10

### PR DESCRIPTION
**Pull request by mzaian**
Cherry pick of #121566 on release-1.27.

#121566: etcd: Update to version 3.5.10

**Does this PR introduce a user-facing change?**:

```release-note
etcd: Update to version 3.5.10
```

### CODE REVIEW

File: build/dependencies.yaml [hunk 1]
```diff
@@ -62,7 +62,7 @@ dependencies:
 
   # etcd
   - name: "etcd"
-    version: 3.5.9
+    version: 3.5.10
```
Comment (jmhbnz) [inline build/dependencies.yaml:65]: Should we also be updating the etcd image on line 83 below?

**Review by jmhbnz — COMMENTED** [review]
Hey @mzaian - Just checking I can see a few more references to `3.5.9` in `release-1.27`.  Should we be updating these also?

```bash
 james  ~  Documents  kubernetes   automated-cherry-pick-..                                                                                                                    14:22:15 
 ➜ grep -Ri '3\.5\.9'
build/dependencies.yaml:    version: 3.5.9
cluster/images/etcd/migrate/options.go: supportedEtcdVersions = []string{"3.0.17", "3.1.20", "3.2.32", "3.3.17", "3.4.18", "3.5.9"}
cluster/images/etcd/Makefile:BUNDLED_ETCD_VERSIONS?=3.0.17 3.1.20 3.2.32 3.3.17 3.4.18 3.5.9
```

File: build/dependencies.yaml (hunk 1 above)
Comment (mzaian) [reply build/dependencies.yaml:65]: I just did that :)

**Review by jmhbnz — APPROVED** [review]
/approve
/lgtm

Thanks @mzaian

**Review by justaugustus — APPROVED** [review]
/lgtm
/approve
/assign @liggitt

**Review by jmhbnz — APPROVED** [review]
/lgtm

**Review by xmudrii — APPROVED** [review]
For RelEng:
/lgtm
/approve

**Pull request merged**